### PR TITLE
Fixes from lesson03

### PR DESCRIPTION
**Pull request by JohnSkull**
Hi Natasha, I ran some code analysis and hopefully this is more in line with code quality expectations for the exercises. Additionally I added the sorted function, and fixed a couple of errors that you pointed out. Thank you.

### CODE REVIEW

File: students/john_hunter/lesson03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    """Returns a report of the donors by total historical amount
+    The report should contain the Donor Name, total donated, number of
+    donations, and average donation amount as values in each row
+   """
+    max_len_of_name = 0
+    donors_temp_two = donors
+
+    for i in range(len(donors_temp_two)):#calculates the longest name for formating
+        if len(donors_temp_two[i][0]) > max_len_of_name:
+            max_len_of_name = len(donors_temp_two[i][0])
```
Comment (natasha-aleksandrova) [inline students/john_hunter/lesson03/mailroom.py:156]: this is a bit overcomplicated :
```

for donor in donors:
    name = donor[0]
    if len(name) > max_len_of_name:
        max_len_of_name = name
```

no need for temp variable or using range() function.

File: students/john_hunter/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            break
+        else:
+            int(donation)
+            donations.insert(-1, donation)
+    for i in range(len(donors)):
+        if name == donors[i][0]:
+            donors[i] = donors[i] + donations
+            break
+    print(donors[i])
+    return sum(donors[i][1:])
```
Comment (natasha-aleksandrova) [inline students/john_hunter/lesson03/mailroom.py:144]: same here, range() is unnecessary


instead
```
for donor in donors:
    if donor[0] == name:
        donor.append(donation)
        break
else:
    donors.append([name, donation])
```

File: students/john_hunter/lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        more_donations = input(mas_donations)
+    if more_donations == 'y':
+        totals = add_donations(input_name)
+    else:
+        totals = 0
+        for z in range(len(donors)):
+            if input_name == donors[z][0]:
+                for c in range(len(donors[z])-1):
+                    totals = totals + int(donors[z][c+1])
+                break
```
Comment (natasha-aleksandrova) [inline students/john_hunter/lesson03/mailroom.py:89]: no need to use range() here and no need to loop to get totals.

utilize slice syntax to get list of donations, then use sum() function

**Review by natasha-aleksandrova — COMMENTED** [review]
left some feedback to use python best practices when looping over lists and updating your list.

**Comment by natasha-aleksandrova** [comment]
@JohnSkull github shows conflicts which will prevent me from being able to merge.

did you use two different computers? let me know if we can help to resolve conflicts.

**Comment by JohnSkull** [comment]
I'm going to do a git push --force origin master to overwrite the merge conflicts. I will close this and raise a new request.

**Pull request closed without merging**